darwin/DarwinMachine.c

```diff
@@ -24,6 +25,12 @@ in the source distribution for its full text.
 #include "generic/openzfs_sysctl.h"
 #include "zfs/ZfsArcStats.h"
 
+// Compatibility for older SDKs.
+#if __IPHONE_OS_VERSION_MIN_REQUIRED <= __IPHONE_6_1
+#ifndef kIOMainPortDefault
```

Comment:
This compatibility definition shouldn't be limited to iOS. (The internet says some older macOS versions also need this.)

And, because `kIOMainPortDefault` is defined as a constant variable, not a macro, a better way to check it is to use `AC_CHECK_DECLS` in `configure`.

```m4
AC_CHECK_DECLS([kIOMainPortDefault], [], [], [[#include <IOKit/IOKitLib.h>]])
```

```c
#if defined(HAVE_DECL_KIOMAINPORTDEFAULT) && !HAVE_DECL_KIOMAINPORTDEFAULT
#define kIOMainPortDefault kIOMasterPortDefault
#endif
```

This way would work better.